x-pack/plugins/security/public/management/roles/edit_role/privileges/kibana/privilege_summary/privilege_summary_calculator.ts

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Role, RoleKibanaPrivilege } from '../../../../../../../common/model';
+import { isGlobalPrivilegeDefinition } from '../../../privilege_utils';
+import { KibanaPrivileges, PrimaryFeaturePrivilege, SecuredFeature } from '../../../../model';
+import { PrivilegeCollection } from '../../../../model/privilege_collection';
+
+export interface EffectiveFeaturePrivileges {
+  [featureId: string]: {
+    primary?: PrimaryFeaturePrivilege;
+    subFeature: string[];
+    hasNonSupersededSubFeaturePrivileges: boolean;
+  };
+}
+export class PrivilegeSummaryCalculator {
+  constructor(private readonly kibanaPrivileges: KibanaPrivileges, private readonly role: Role) {}
+
+  public getEffectiveFeaturePrivileges(entry: RoleKibanaPrivilege): EffectiveFeaturePrivileges {
```

Comment:
Similar to what was noted elsewhere, when Discover "all" is assigned, customize sub-feature privileges is selected, and "Create Short URLs" is deselected, the information tooltip isn't being displayed.